### DISCUSSION TITLE
Different speed settings for fade and schemes

### DISCUSSION THREAD

**Discussion by Rapti** [arendst/Tasmota#12329] [Ideas]
I'm using Tasmota to control PWM RGB strips. Sometimes I want a static color, but sometimes I like to use the rainbow effect provided by `scheme 2`. Unfortunately, both the fade effect when I change the static color and the rainbow effect use the same `speed` setting. I'd like to have a short transition time between static color changes on one hand and a nice and slow rainbow effect on the other hand.
I kind of worked around this problem using some weird hacky rules on OpenHAB's side, but this is of course far from optimal and I think Tasmota should have native support for that.

Is that already possible and I'm missing something? If not, please add separate speed settings for `fade` and effect schemes.

**Comment by sfromis** [comment]
As schemes "just" use fading, it is indeed the same speed option. You could reassign speed when you activate the commands for schemes, color changes or power on/off.

Adding features usually works best if someone does the changes and submits a PR ;-)

**Reply by Rapti** [reply]
Yeah, that's what I'm doing right now. But that's rather hacky and it's only possible when controlling Tasmota through an external automation service.

**Reply by sfromis** [reply]
Nothing stops you from also making it work for local physical buttons, just decouple from the light, and create rules to control the flow. You could wrap it in `Event` commands, allowing the same commands to work from rules or from external sources.
